### DISCUSSION TITLE
Hierarchical view is not supported by this JSON-LD processor

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#5164] [General]
My code is not working any more. Maybe because of upgrading to 5.0.2 ?

```
		WriterConfig config = getWriterConfig();
		StringWriter stringWriter = new StringWriter();
		Rio.write(input, stringWriter, RDFFormat.JSONLD, config);
		String string = stringWriter.toString();

```
```
private WriterConfig getWriterConfig() {
		WriterConfig config = new WriterConfig();
		config.set(JSONLDSettings.HIERARCHICAL_VIEW, true);
		config.set(JSONLDSettings.COMPACT_ARRAYS, true);
		config.set(JSONLDSettings.OPTIMIZE, true);
		return config;
	}
```

Does that mean, i canntot use `Rio.write` any more ?

```
stacktrace: java.lang.UnsupportedOperationException: Hierarchical view is not supported by this JSON-LD processor. Use org.eclipse.rdf4j.rio.jsonld.legacy.JSONLDWriter instead.
	at org.eclipse.rdf4j.rio.jsonld.JSONLDWriter.endRDF(JSONLDWriter.java:154)
	at org.eclipse.rdf4j.rio.Rio.write(Rio.java:653)
	at org.eclipse.rdf4j.rio.Rio.write(Rio.java:608)
```

**Comment by hmottestad** [comment]
You'll have to use the legacy one.

**Reply by naturzukunft** [reply]
i'm searching the dependency

**Reply by hmottestad** [reply]
https://mvnrepository.com/artifact/org.eclipse.rdf4j/rdf4j-rio-jsonld-legacy

**Reply by hmottestad** [reply]
Make sure to exclude the new one from any dependencies that include it as a transitive dependency.

**Reply by naturzukunft** [reply]
seems to work, but does not feel good to use legacy things ;-o
i started a complete build and waiting for the next issue ;-)

**Reply by naturzukunft** [reply]
Ok, looks good. The project has been built, no tests are failed.
I'm using now 

- spring-boot: 3.3.5
- rdf4j: 5.0.2 (model api and repository)

but without the sparql endpoint (rdf4j-spring-boot-sparql-web).

**Reply by hmottestad** [reply]
I would recommend looking into framing with JSON-LD 1.1 instead. We should have an example or two in the unit tests. 

Framing is a great way of making json-ld more accessible for developers who are used to json. It's not as generic as the hierarchical jsond-ld, but it's a lot more supported and standardised. Also a lot more powerful.

**Reply by naturzukunft** [reply]
is framing the right solution for generic results? i didn't know the structure, that i have to format.

**Reply by hmottestad** [reply]
Not for a generic result no.

What is your use case for hierarchical json-ld?

**Reply by naturzukunft** [reply]
Presenting rdf in a way that is understood by as many [activity-pub](https://www.w3.org/TR/activitypub/) implementierungen as possible.

I may have to make certain individual translations, depending on which implementation I am communicating with. That remains to be seen. From what I've already learnt, interoperability is not that great.

With individual translations, framing per implementation and possibly type may become interesting.

**Comment by naturzukunft** [comment]
at the moment i stumble over this error 

```
java.lang.NoSuchMethodError: 'java.lang.Object com.github.jsonldjava.utils.JsonUtils.fromJsonParser(com.fasterxml.jackson.core.JsonParser)'
	at org.eclipse.rdf4j.rio.jsonld.legacy.JSONLDParser.getJSONObject(JSONLDParser.java:153) ~[rdf4j-rio-jsonld-legacy-5.0.2.jar:5.0.2]
	at org.eclipse.rdf4j.rio.jsonld.legacy.JSONLDParser.parse(JSONLDParser.java:133) ~[rdf4j-rio-jsonld-legacy-5.0.2.jar:5.0.2]
	at org.eclipse.rdf4j.rio.jsonld.legacy.JSONLDParser.parse(JSONLDParser.java:103) ~[rdf4j-rio-jsonld-legacy-5.0.2.jar:5.0.2]
```

**Reply by naturzukunft** [reply]
probably maven decided to use the wrong version o
```
			<groupId>com.github.jsonld-java</groupId>
			<artifactId>jsonld-java</artifactId>

```

**Reply by hmottestad** [reply]
Probably a version conflict yes. IntelliJ has a nice diagram tool to show version conflicts in maven.
